### ISSUE TITLE
fetchAddressBooks results in all entities being listed

### ISSUE THREAD

**Issue by chmac** [natelindev/tsdav#74]
I'm using tsdav to connect to xandikos. During the `fetchAddressBooks() call, I see the following request being sent to xandikos:

```xml
 <?xml version="1.0" encoding="utf-8"?>
 <d:propfind xmlns:c="urn:ietf:params:xml:ns:caldav" xmlns:ca="http://apple.com/ns/ical/" xmlns:cs="http://calendarserver.org/ns/" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:d="DAV:">
   <d:prop>
     <d:supported-report-set/>
   </d:prop>
 </d:propfind>
```

And I see the following response back:

```xml
<ns0:multistatus xmlns:ns0="DAV:" xmlns:ns1="urn:ietf:params:xml:ns:carddav">
  <ns0:response>
    <ns0:href>/user/contacts/addressbook/</ns0:href>
    <ns0:propstat>
      <ns0:status>HTTP/1.1 200 OK</ns0:status>
      <ns0:prop>
        <ns0:supported-report-set>
          <ns0:supported-report>
            <ns1:addressbook-multiget />
          </ns0:supported-report>
          <ns0:supported-report>
            <ns1:addressbook-query />
          </ns0:supported-report>
          <ns0:supported-report>
            <ns0:expand-property />
          </ns0:supported-report>
          <ns0:supported-report>
            <ns0:sync-collection />
          </ns0:supported-report>
        </ns0:supported-report-set>
      </ns0:prop>
    </ns0:propstat>
  </ns0:response>
  <ns0:response>
    <ns0:href>/user/contacts/addressbook/913e5a13-454c-e34a-945e-1354c9c5e92b.vcf</ns0:href>
    <ns0:propstat>
      <ns0:status>HTTP/1.1 404 Not Found</ns0:status>
      <ns0:prop>
        <ns0:supported-report-set />
      </ns0:prop>
    </ns0:propstat>
  </ns0:response>
  <ns0:response>
    <ns0:href>/user/contacts/addressbook/923e5a13-454c-e34a-945e-1354c9c5e92b.vcf</ns0:href>
    <ns0:propstat>
      <ns0:status>HTTP/1.1 404 Not Found</ns0:status>
      <ns0:prop>
        <ns0:supported-report-set />
      </ns0:prop>
    </ns0:propstat>
  </ns0:response>
  <ns0:response>
    <ns0:href>/user/contacts/addressbook/9479c4c0-ec6a-7b4d-87b4-3e0e07a55f1b.vcf</ns0:href>
    <ns0:propstat>
      <ns0:status>HTTP/1.1 404 Not Found</ns0:status>
      <ns0:prop>
        <ns0:supported-report-set />
      </ns0:prop>
    </ns0:propstat>
  </ns0:response>
  <ns0:response>
    <ns0:href>/user/contacts/addressbook/abc.vcf</ns0:href>
    <ns0:propstat>
      <ns0:status>HTTP/1.1 404 Not Found</ns0:status>
      <ns0:prop>
        <ns0:supported-report-set />
      </ns0:prop>
    </ns0:propstat>
  </ns0:response>
  <ns0:response>
    <ns0:href>/user/contacts/addressbook/abcd.vcf</ns0:href>
    <ns0:propstat>
      <ns0:status>HTTP/1.1 404 Not Found</ns0:status>
      <ns0:prop>
        <ns0:supported-report-set />
      </ns0:prop>
    </ns0:propstat>
  </ns0:response>
</ns0:multistatus>
```

I've tried the same against FastMail, and it also lists every record in the account. I'm quite new to the whole WebDAV space, and CardDAV, so I might be wrong here, but isn't there an option to find "address books" without pulling every record. I notice that in the response tsdav says `res[0]`, so it looks like every item is being listed but only the first item is being used.

Of course, maybe that's just the way WebDAV works and there's no way round it, in which case feel free to close this issue without comment.

**Comment by natelindev** [comment]
Might be a bug, I'll take a look next week

**Comment by natelindev** [comment]
Turns out this is a bug, I mistakenly used `depth 1` in the header of the request, after changing it to `depth 0`, the response become something like 

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<multistatus xmlns="DAV:">
    <response xmlns="DAV:">
        <href>/12345/calendars/12345/</href>
        <propstat>
            <prop>
                <supported-report-set xmlns="DAV:">
                    <supported-report>
                        <report>
                            <acl-principal-prop-set></acl-principal-prop-set>
                        </report>
                    </supported-report>
                    <supported-report>
                        <report>
                            <principal-match></principal-match>
                        </report>
                    </supported-report>
                    <supported-report>
                        <report>
                            <principal-property-search></principal-property-search>
                        </report>
                    </supported-report>
                    <supported-report>
                        <report>
                            <sync-collection></sync-collection>
                        </report>
                    </supported-report>
                    <supported-report>
                        <report>
                            <calendar-query xmlns="urn:ietf:params:xml:ns:caldav"></calendar-query>
                        </report>
                    </supported-report>
                    <supported-report>
                        <report>
                            <calendar-multiget xmlns="urn:ietf:params:xml:ns:caldav"></calendar-multiget>
                        </report>
                    </supported-report>
                    <supported-report>
                        <report>
                            <free-busy-query xmlns="urn:ietf:params:xml:ns:caldav"></free-busy-query>
                        </report>
                    </supported-report>
                    <supported-report>
                        <report>
                            <calendar-search xmlns="http://calendarserver.org/ns/"></calendar-search>
                        </report>
                    </supported-report>
                </supported-report-set>
            </prop>
            <status>HTTP/1.1 200 OK</status>
        </propstat>
    </response>
</multistatus>
```

which is exactly what we needed.

This will be fixed in upcoming 2.0.0-rc.2
